include/deal.II/grid/grid_tools.h

```diff
@@ -4035,6 +4056,233 @@ namespace GridTools
           MPI_Waitall(requests.size(), requests.data(), MPI_STATUSES_IGNORE);
         AssertThrowMPI(ierr);
       }
+#    endif // DEAL_II_WITH_MPI
+  }
+
+
+
+  template <typename DataType, typename MeshType>
+  void
+  exchange_cell_data_to_level_ghosts(
+    const MeshType &                                    mesh,
+    const std::function<std_cxx17::optional<DataType>(
+      const typename MeshType::level_cell_iterator &)> &pack,
+    const std::function<void(const typename MeshType::level_cell_iterator &,
+                             const DataType &)> &       unpack,
+    const std::function<bool(const typename MeshType::level_cell_iterator &)>
+      &filter)
+  {
+#    ifndef DEAL_II_WITH_MPI
+    (void)mesh;
+    (void)pack;
+    (void)unpack;
+    (void)filter;
+    Assert(false,
+           ExcMessage(
+             "GridTools::exchange_cell_data_to_ghosts() requires MPI."));
+#    else
+    constexpr int dim      = MeshType::dimension;
+    constexpr int spacedim = MeshType::space_dimension;
+    auto          tria =
+      dynamic_cast<const parallel::TriangulationBase<dim, spacedim> *>(
+        &mesh.get_triangulation());
+    Assert(
+      tria != nullptr,
+      ExcMessage(
+        "The function exchange_cell_data_to_ghosts() only works with parallel triangulations."));
+
+    // build list of cells to request for each neighbor
+    std::set<dealii::types::subdomain_id> level_ghost_owners =
+      tria->level_ghost_owners();
+    std::map<dealii::types::subdomain_id,
+             std::vector<typename CellId::binary_type>>
+      neighbor_cell_list;
+
+    for (const auto level_ghost_owner : level_ghost_owners)
+      neighbor_cell_list[level_ghost_owner] = {};
+
+    {
+      typename MeshType::level_cell_iterator cell, endc = mesh.end();
+      for (cell = mesh.begin(); cell != endc; ++cell)
+        if (cell->level_subdomain_id() !=
+              dealii::numbers::artificial_subdomain_id &&
+            !cell->is_locally_owned_on_level())
+          if (filter(cell))
+            neighbor_cell_list[cell->level_subdomain_id()].emplace_back(
+              cell->id().template to_binary<spacedim>());
+    }
+
+    Assert(level_ghost_owners.size() == neighbor_cell_list.size(),
+           ExcInternalError());
+
+
+    // Before sending & receiving, make sure we protect this section with
+    // a mutex:
+    static Utilities::MPI::CollectiveMutex      mutex;
+    Utilities::MPI::CollectiveMutex::ScopedLock lock(mutex,
+                                                     tria->get_communicator());
+
+    const int mpi_tag =
+      Utilities::MPI::internal::Tags::dofhandler_communicate_mg_ghost_cells;
```

Comment:
Are you planning to update the mpi tag later to a similar name we have for the exchange function on the active cells or is this simply an oversight?